src/mon/OSDMonitor.cc

```diff
@@ -2208,23 +2207,27 @@ bool OSDMonitor::check_failure(utime_t now, int target_osd, failure_info_t& fi)
   // help us localize the grace correction to a subset of the system
   // (say, a rack with a bad switch) that is unhappy.
   assert(fi.reporters.size());
-  for (map<int,failure_reporter_t>::iterator p = fi.reporters.begin();
-	p != fi.reporters.end();
-	++p) {
+  for (auto p = fi.reporters.begin(); p != fi.reporters.end();) {
     // get the parent bucket whose type matches with "reporter_subtree_level".
     // fall back to OSD if the level doesn't exist.
-    map<string, string> reporter_loc = osdmap.crush->get_full_location(p->first);
-    map<string, string>::iterator iter = reporter_loc.find(reporter_subtree_level);
-    if (iter == reporter_loc.end()) {
-      reporters_by_subtree.insert("osd." + to_string(p->first));
+    if (osdmap.exists(p->first)) {
+      auto reporter_loc = osdmap.crush->get_full_location(p->first);
+      if (auto iter = reporter_loc.find(reporter_subtree_level);
```

Comment:
```
/home/jenkins-build/build/workspace/ceph-pull-requests/src/mon/OSDMonitor.cc:2215:64: error: expected ‘)’ before ‘;’ token
       if (auto iter = reporter_loc.find(reporter_subtree_level);
                                                                ^
```

i think luminous is not using C++17. probably we can move 
```c++
auto iter = reporter_loc.find(reporter_subtree_level);
```
out of the `if` statement.